### PR TITLE
WT-6912 Architecture Guide - logging page.

### PR DESCRIPTION
**Pull request by sueloverso**
This is ready for the first round of review.

### CODE REVIEW

File: src/docs/arch-logging.dox [hunk 1]
```diff
@@ -1,8 +1,167 @@
...
+A log sequence number (LSN) identifies the location of a log record. In WiredTiger
+LSNs is a 64-bit number containing two values, a 32-bit file number and a 32-bit offset.
+The file number portion of the LSN maps to the number in the \c WiredTigerLog. file name.
+The offset portion of the LSN is the byte offset within that file where the log record starts.
+
+There is a \c WT_LOG structure that contains the live information about the state of
+the logging subsystem. This document does not itemize every field. It contains some
+locks and condition variables for synchronization and a large set of LSNs for keeping
+track of the state of the system. Some of the important LSNs maintained are:
+* alloc_lsn: This is the LSN for the next log record allocation.
```
Comment (tammybailey) [inline src/docs/arch-logging.dox:80]: Capitalisation of "this" is inconsistent across the four items

File: src/docs/arch-logging.dox [hunk 2]
```diff
@@ -1,8 +1,167 @@
...
+create fewer of them.
+
+There is another internal thread dedicated to managing operations on log files. This
+is the \c log_file_server thread. I/O operations can be slow and costly so this
+thread takes the burden of closing and syncing files to disk off of user application threads.
+If a log file needs to be closed, then the application thread moves its file handle to the
+\c log->log_close_fh field and signals this thread. This thread will fsync the file
+handle, and then close it. Updating the \c log_close_lsn and the \c sync_lsn values.
+
+The third internal thread manages advancing the \c write_lsn in proper LSN order.
```
Comment (tammybailey) [inline src/docs/arch-logging.dox:131]: What does proper LSN order mean? I feel like the second sentence repeats what you said in the first. Would it make sense to rewrite as "The third internal thread manages advancing the `write_lsn`. As the `write_lsn` must be advanced in proper LSN order, it becomes a point of single threading in the path of writing a log record."

File: src/docs/arch-logging.dox [hunk 3]
```diff
@@ -1,8 +1,167 @@
...
+will coalesce adjacent LSNs of slots in order to advance the LSN in larger increments.
+This thread will then clean up and mark the state of the processed slots
+as \c WT_LOG_SLOT_FREE.
+
+@subsection Using the log
+
+In addition to storing a record of the operations performed to data in the write-ahead
+log, users can write their own strings into the logs via the \c WT_SESSION::log_printf
+method. That method takes a printf-style format string and arguments and will write
+that string into the write-ahead log as a message record type.
```
Comment (tammybailey) [inline src/docs/arch-logging.dox:147]: Does message record type mean `WT_LOGREC_MESSAGE`?

File: src/docs/arch-logging.dox [hunk 4]
```diff
@@ -1,8 +1,167 @@
...
+
+Transactions are stored in the write-ahead log and their purpose is to allow recovery
+in the case of crash or failure. When the system is restarted, WiredTiger will always
+run recovery. If the system previously shutdown cleanly, then recovery determines that
+the log does not need to be replayed. Otherwise, recovery replays all operations from
+the LSN of the most recent checkpoint. When timestamps are in use, recovery also
+performs a rollback-to-stable operation on tables that are not logged.
+# Remove this line and uncomment the next line after the rollback to stable page exists.
```
Comment (tammybailey) [inline src/docs/arch-logging.dox:155]: [This page exists](https://github.com/wiredtiger/wiredtiger/blob/develop/src/docs/arch-rts.dox).

**Review by tammybailey — CHANGES_REQUESTED** [review]

File: src/docs/arch-logging.dox [hunk 5]
```diff
@@ -1,8 +1,167 @@
...
 
-WiredTiger writes all changes into a write-ahead log when configured.
-Log files are shared among all writers, so changes made to
-different tables, by different threads, are interleaved.
+The WiredTiger logging subsystem implements a write-ahead log when configured.
+It is a central resource recording changes made to different tables, by different
+threads, into a file on disk. Its purpose is so that changes made since the most recent
+checkpoint are not lost and can be recovered on restart after a crash (unclean shutdown).
+
+@subsection Log-related files in the database directory
```
[the remark this answers is not in the document]
Comment (sueloverso) [reply src/docs/arch-logging.dox]: Thanks fixed.

File: src/docs/arch-logging.dox [hunk 6]
```diff
@@ -1,8 +1,167 @@
...
+The other two log-related filenames are for log file pre-allocation. When pre-allocation
+is configured (the default), new log files are created ahead of time containing the
+log file header content and are ready in the directory for use. The header content is
+written into a temporary file with the prefix \c WiredTigerTmplog. and again followed
+by a number. Once the temporary file is completely filled with the required content,
+synced to disk and closed, it is renamed to a prepared log file that other threads in
+the system can rename to use as the next real log file. Prepared pre-allocated log files
+that are ready to use have the prefix \c WiredTigerPreplog.
+
+There varying number of temporary and prepared log files in the directory. They
```
[the remark this answers is not in the document]
Comment (sueloverso) [reply src/docs/arch-logging.dox:22]: FYI, as I go through your comments, if I just accept it, I'm going to resolve it as I make my edit and not type "Fixed" into every single box. I'll only reply if I disagree for some reason.

File: src/docs/arch-logging.dox [hunk 7]
```diff
@@ -1,8 +1,182 @@
...
+@section log_configuration Configuration choices and their implications
+
+Once the logging subsystem is enabled for the system, then other configuration settings
+control the finer details of the logging subsystem. The \c archive=true setting turns
+on log file archiving. That means that log files earlier than the most recently
+completed checkpoint can be removed. (In this use, archive means removal not a form
+of cold storage.) There are times, even when configured on, that log archiving is
+temporarily disabled inside the WiredTiger library.
+
+Log archiving is disabled when a backup cursor is open, WiredTiger must guarantee
```
Comment (tammybailey) [inline src/docs/arch-logging.dox:43]: I think the `','` after `open` should be a `'.'`.

**Review by tammybailey — APPROVED** [review]
One small punctuation comment, otherwise LGTM!

**Pull request merged**